File: java/src/java-mirror-gen/tests/testData/cj_mirror_gen_str_param/M.java
// Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
// This source file is part of the Cangjie project, licensed under Apache-2.0
// with Runtime Library Exception.
//
// See https://cangjie-lang.cn/pages/LICENSE for license information.

import java.lang.String;

public class M {
    public int f;

    public M(int f) {
        this.f = f;
    }

    public void goo(String arg) {
        System.out.println(arg);
    }

    public void gooMultipleArgs(String arg1, int arg2) {
        System.out.println(arg1);
        System.out.println(arg2);
    }
}

File: objc/src/interoplib/examples/objc_pointer/origin-objc/M.m
// Copyright (c) Huawei Technologies Co., Ltd. 2025. All rights reserved.
// This source file is part of the Cangjie project, licensed under Apache-2.0
// with Runtime Library Exception.
//
// See https://cangjie-lang.cn/pages/LICENSE for license information.

#import "M.h"

double PI = 3.14;
double* pPI = &PI;

M* global;

int64_t primGlobal = 56;

@implementation M

- (void*) pointerToUnit {
    return (__bridge void*)self;
}
- (int64_t*) pointerToPrimitive {
    return &primGlobal;
}
- (double**) pointerToPointer {
    return &pPI;
}
- (M*__strong*) pointerToMirror {
    global = self;
    return &global;
}
- (M*__strong*) pointerToImpl {
    global = self;
    return &global;
}
- (double) baseline {
    return 3.16;
}
- (void) run {
    printf("M::run\n");
}

@end
